Source/OxyPlot/PlotModel/PlotModel.Rendering.cs

```diff
@@ -500,7 +500,8 @@ private void UpdatePlotArea(IRenderContext rc)
 
             // Find the available size for the legend box
             var availableLegendWidth = plotArea.Width;
```

Comment:
I think legend items that don't fit within the max width/height should be skipped. Not sure if that works, will test!
I see there is a lot of cases to test.. Will play with the new code! 
It sounds good not to change the current behavior.